src/jsifier.js

```diff
@@ -228,11 +228,11 @@ function(${args}) {
     if (SHARED_MEMORY) {
       const proxyingMode = LibraryManager.library[symbol + '__proxy'];
       if (proxyingMode) {
-        if (proxyingMode !== 'sync' && proxyingMode !== 'async') {
-          throw new Error(`Invalid proxyingMode ${symbol}__proxy: '${proxyingMode}' specified!`);
-        }
-        const sync = proxyingMode === 'sync';
         if (PTHREADS) {
+          if (proxyingMode !== 'sync' && proxyingMode !== 'async') {
+            error(`Invalid proxyingMode ${symbol}__proxy: '${proxyingMode}' specified!`);
+          }
+          const sync = proxyingMode === 'sync';
```

Comment:
I'm missing something obvious. There is no `WASM_WORKERS` around this changed code. Why does the change only affect that mode?

Reply:
`WASM_WORKERS` defines `SHARED_MEMORY` but not `PTHREADS`.

The code in `src/library_html5_webgl.js` that re-writes the custom `__proxy` all behind `#if PTHREADS` so when `WASM_WORKERS` is defined we end up getting here but with custom `__proxy` stuff.